x-pack/plugins/rule_registry/server/alert_data_client/tests/update.test.ts

```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { AlertsClient, ConstructorOptions } from '../alerts_client';
+import { loggingSystemMock } from '../../../../../../src/core/server/mocks';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { elasticsearchClientMock } from 'src/core/server/elasticsearch/client/mocks';
+import { ruleDataPluginServiceMock } from '../../rule_data_plugin_service/rule_data_plugin_service.mock';
+import { RuleDataPluginServiceConstructorOptions } from '../../rule_data_plugin_service';
+import { alertingAuthorizationMock } from '../../../../alerting/server/authorization/alerting_authorization.mock';
+import { AuditLogger } from '../../../../security/server';
+
+const ruleDataServiceMock = ruleDataPluginServiceMock.create(
+  {} as RuleDataPluginServiceConstructorOptions
+);
+const alertingAuthMock = alertingAuthorizationMock.create();
+const esClientMock = elasticsearchClientMock.createElasticsearchClient();
+const auditLogger = {
+  log: jest.fn(),
+} as jest.Mocked<AuditLogger>;
+
+const alertsClientParams: jest.Mocked<ConstructorOptions> = {
+  logger: loggingSystemMock.create().get(),
+  ruleDataService: {
+    ...ruleDataServiceMock,
+    getFullAssetName: (_?: string | undefined) => '.alerts-observability-apm',
+  },
+  authorization: alertingAuthMock,
+  esClient: esClientMock,
+  auditLogger,
+};
+
+beforeEach(() => {
+  jest.resetAllMocks();
+});
+
+describe('update()', () => {
```

Comment:
The top level `describe` should probably be the name of the thing, so like:

```
describe('AlertDataClient, () => {
  describe('update', () =>
```

The `()` isn't needed either.

Reply:
Was following alertings conventions as we borrowed from their architecture/file structure. Happy to update if you feel strongly it should be changed :)